cmd/migratecheckpoint/main.go

```diff
@@ -0,0 +1,364 @@
+		Key:  key,
+		Type: Get,
+	}
+}
+
+func DeleteOperation(key string) Operation {
+	return &operation{
+		Key:  key,
+		Type: Delete,
+	}
+}
+
+var defaultBucket = []byte(`default`)
+
+type fileStorageClient struct {
+	db *bbolt.DB
+}
+
+func newClient(filePath string, timeout time.Duration) (*fileStorageClient, error) {
+	options := &bbolt.Options{
+		Timeout: timeout,
+		NoSync:  true,
+	}
+	db, err := bbolt.Open(filePath, 0600, options)
+	if err != nil {
+		return nil, err
+	}
+
+	initBucket := func(tx *bbolt.Tx) error {
+		_, err := tx.CreateBucketIfNotExists(defaultBucket)
+		return err
+	}
+	if err := db.Update(initBucket); err != nil {
+		return nil, err
+	}
+
+	return &fileStorageClient{db}, nil
+}
+
+// Set will store data. The data can be retrieved using the same key
+func (c *fileStorageClient) Set(key string, value []byte) error {
+	return c.Batch(SetOperation(key, value))
+}
+
+// Batch executes the specified operations in order. Get operation results are updated in place
+func (c *fileStorageClient) Batch(ops ...Operation) error {
+	batch := func(tx *bbolt.Tx) error {
+		bucket := tx.Bucket(defaultBucket)
+		if bucket == nil {
+			return errors.New("storage not initialized")
+		}
+
+		var err error
+		for _, op := range ops {
+			switch op.Type {
+			case Get:
+				op.Value = bucket.Get([]byte(op.Key))
+			case Set:
+				err = bucket.Put([]byte(op.Key), op.Value)
+			case Delete:
+				err = bucket.Delete([]byte(op.Key))
+			default:
+				return errors.New("wrong operation type")
+			}
+
+			if err != nil {
+				return err
+			}
+		}
+
+		return nil
+	}
+
+	return c.db.Update(batch)
+}
+
+// Close will close the database
+func (c *fileStorageClient) Close() error {
+	return c.db.Close()
+}
+
+func getEnv(key, fallback string) string {
+	if value, ok := os.LookupEnv(key); ok {
+		return value
+	}
+	return fallback
+}
+
+func main() {
+	containerLogPathFluentd := getEnv("CONTAINER_LOG_PATH_FLUENTD", "/var/log/splunk-fluentd-containers.log.pos")
+	containerLogPathOtel := getEnv("CONTAINER_LOG_PATH_OTEL", "/var/lib/otel_pos/receiver_filelog_")
+
+	customLogPathFluentd := getEnv("CUSTOM_LOG_PATH_FLUENTD", "/var/log/splunk-fluentd-*.pos")
+	customLogPathOtel := getEnv("CUSTOM_LOG_PATH_OTEL", "/var/lib/otel_pos/receiver_filelog_")
+
+	// Check whether it has already Otel's checkpoints
+	_, err := os.Stat(containerLogPathOtel)
+	if !os.IsNotExist(err) {
+		// Otel checkpoint already present. no need to migrate.
+		return
+	}
+
+	// Check whether there are fluentd's position file exist
+	_, err = os.Stat(containerLogPathFluentd)
+	if os.IsNotExist(err) {
+		// Fluentd position file does not exist. no need to migrate.
+		return
+	}
+
+	// Container File Logs
+	lines, err := readLines(containerLogPathFluentd)
+	if err != nil {
+		fmt.Println("Error reading container fluentd's log path")
+		panic(err)
+	}
+
+	var readers []*Reader
+	var reader *Reader
+
+	client, err := newClient(containerLogPathOtel, 100)
+	if err != nil {
+		fmt.Println(err)
```

Comment:
Let's add a non-zero exit here or panic in order to avoid going further with nil client